src/tictactoe_clojure/computer_player.clj

```diff
@@ -0,0 +1,91 @@
+(ns tictactoe-clojure.computer-player
+  (:require [tictactoe-clojure.console :as console])
+  (:require [tictactoe-clojure.board :as board])
+  (:use clojure.set)
+  (:require [tictactoe-clojure.rules :as rules]))
+
+(defn- create-pair
+  [coll-1 coll-2]
+    (partition 2 (interleave coll-1 coll-2)))
+
+(declare optimal-move-by-score)
+(declare score-board)
+
+(def high-value-pos-4x4 #{1 4 13 16})
+(def high-value-pos-5x5 #{13 1 5 21 25})
+
+(defn- score-board 
+  [player board depth maximum]
+    (cond 
+      (= (rules/draw? board) true) 0
+      (= (rules/winner board) player) 
+        (if maximum 
+          (- 10 depth) 
+          (+ -10 depth))
+      (= (rules/winner board) (rules/switch-turn player)) 
+        (if maximum 
+          (+ -10 depth) 
+          (- 10 depth))
+      :else 
+        (second 
+          (optimal-move-by-score (rules/switch-turn player) board (dec depth) (not maximum)))))
+
+(def memo-score-board (memoize score-board))
+
+(defn- optimal-move-by-score 
+  [player board depth maximum]
+    (let [possible-moves (board/valid-slots board)
+          score-of-possible-move (map #(memo-score-board player (board/move board % player) depth maximum) possible-moves)
+          move-score-pair (create-pair possible-moves score-of-possible-move)
+          sorted-move-scores-pair (sort-by second move-score-pair)]
+      (if maximum 
+        (last sorted-move-scores-pair)
+        (first sorted-move-scores-pair))))
+
+(defn- check-seq-for-dups [seq]
+  (for [[id freq] (frequencies seq) 
+        :when (> freq 1)]            
+   id)) 
+
+(defn- win-combo-index-with-dup
+  [board player]
+    (.indexOf 
+      (map #(check-seq-for-dups %) (rules/winning-combinations board)) 
+        (list (rules/switch-turn player))))
+
+(defn- board-sec-with-dup
+  [board player]
+    (let [sec-board-w-dup (win-combo-index-with-dup board player)]
+        (if (not= sec-board-w-dup -1) 
+          sec-board-w-dup 
+          nil)))    
+
+(defn- opponent-close-to-win?
+  [board player]
+    (if (not= nil (board-sec-with-dup board player))
+      true
+      false))
+
+(defn- high-value-move
+  [board valued-moves]
+    (first 
+      (clojure.set/intersection 
+        (set (board/valid-slots board)) 
+        valued-moves)))
+
+(defn- block-opponent-5x5
+  [board player]
+    (first 
+      (board/valid-slots 
+        (nth (rules/winning-combinations board) (board-sec-with-dup board player)))))
+
+(defn optimal-move 
+  [player board depth]
+    (cond
+      (and (> (count (board/valid-slots board)) 12) (= (board/board-size board) 4))
+        (high-value-move board high-value-pos-4x4)
+      (and (= (board/board-size board) 5) (> (count (board/valid-slots board)) 20) (not= nil board-sec-with-dup board player)) (high-value-move board high-value-pos-5x5)
```

Comment:
long lines like this make it very hard to read your code. can you refactor this so that it's more readable? I would suggest making variables or functions that is well named